File: app/src/main/java/com/demeth/massaudioplayer/frontend/components/SquareImageButton.java
package com.demeth.massaudioplayer.frontend.components;

import android.content.Context;
import android.util.AttributeSet;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;

/**
 * This class create a button component that always stay square and that can load an image.
 */
public class SquareImageButton extends androidx.appcompat.widget.AppCompatImageButton {
    public SquareImageButton(@NonNull Context context) {
        super(context);
    }

    public SquareImageButton(@NonNull Context context, @Nullable AttributeSet attrs) {
        super(context, attrs);
    }

    public SquareImageButton(@NonNull Context context, @Nullable AttributeSet attrs, int defStyleAttr) {
        super(context, attrs, defStyleAttr);
    }

    /**
     * resize the button to square view
     * @param widthMeasureSpec width measured
     * @param heightMeasureSpec height measured
     */
    @Override
    protected void onMeasure(int widthMeasureSpec, int heightMeasureSpec) {
        int maxes = Math.max(widthMeasureSpec,heightMeasureSpec);

        //super.onMeasure(maxes,maxes);
        setMeasuredDimension(maxes, maxes);
    }
}

File: app/src/main/java/com/demeth/massaudioplayer/frontend/components/SquareImageButton.kt
package com.demeth.massaudioplayer.frontend.components

import android.content.Context
import android.util.AttributeSet
import androidx.appcompat.widget.AppCompatImageButton


/**
 * This class create a button component that always stay square and that can load an image.
 */
class SquareImageButton(context: Context, attrs: AttributeSet?) : AppCompatImageButton(context,attrs) {


    /**
     * resize the button to square view
     * @param widthMeasureSpec width measured
     * @param heightMeasureSpec height measured
     */
    override fun onMeasure(widthMeasureSpec: Int, heightMeasureSpec: Int) {
        val maxes = widthMeasureSpec.coerceAtLeast(heightMeasureSpec)

        //super.onMeasure(maxes,maxes);
        setMeasuredDimension(maxes, maxes)
    }
}
